internal/charger/shelly.go

```diff
@@ -0,0 +1,132 @@
+package charger
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/andig/evcc/api"
+	"github.com/andig/evcc/util"
+	"github.com/andig/evcc/util/request"
+)
+
+// Shelly api homepage
+// https://shelly-api-docs.shelly.cloud/#common-http-api
+
+// shellyRelayResponse provides the evcc shelly charger enabled information
+type shellyRelayResponse struct {
+	Ison bool `json:"ison,omitempty"`
+}
+
+// shellyStatusResponse provides the evcc shelly charger current power information
+type shellyStatusResponse struct {
+	Meters []struct {
+		Power float64 `json:"power,omitempty"`
+	} `json:"meters,omitempty"`
+}
+
+// Shelly charger implementation
+type Shelly struct {
+	*request.Helper
+	uri          string
+	channel      int
+	standbypower float64
+}
+
+func init() {
+	registry.Add("shelly", NewShellyFromConfig)
+}
+
+// NewShellyFromConfig creates a Shelly charger from generic config
+func NewShellyFromConfig(other map[string]interface{}) (api.Charger, error) {
+	cc := struct {
+		URI          string
+		Channel      int
```

Comment:
gibts einen default channel, z.b. 1?

Reply:
Ja, 0 :-) Hatte es ausprobiert. Wenn in der config der channel nicht gesetzt oder auskommentiert ist, ist die Variable 0, und das ist der erste Channel in der API. Dachte, das dadurch ein explizites Setzen eines Default wertes nicht nötig ist ...

... so Andi mäßig minimalistisch halt ... ;-)

Reply:
top. ich hatte gedanklich mit 1 als 1. Channel gerechnet ;)